src/lib.rs

```diff
@@ -103,31 +105,30 @@ macro_rules! assert_unique_used_features {
     }
 }
 
-assert_unique_used_features! {"llvm4-0", "llvm5-0", "llvm6-0", "llvm7-0", "llvm8-0", "llvm9-0", "llvm10-0", "llvm11-0", "llvm12-0", "llvm13-0", "llvm14-0"}
+assert_unique_used_features! {"llvm4-0", "llvm5-0", "llvm6-0", "llvm7-0", "llvm8-0", "llvm9-0", "llvm10-0", "llvm11-0", "llvm12-0", "llvm13-0", "llvm14-0", "llvm15-0"}
 
-/// Defines the address space in which a global will be inserted.
-///
-/// # Remarks
-/// See also: https://llvm.org/doxygen/NVPTXBaseInfo_8h_source.html
+/// Defines the abstract LLVM address space.
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub enum AddressSpace {
-    Generic = 0,
-    Global = 1,
-    Shared = 3,
-    Const = 4,
-    Local = 5,
+    Zero = 0,
+    One = 1,
+    Two = 2,
+    Three = 3,
+    Four = 4,
+    Five = 5,
```

Comment:
We should retain the original names as they are informative and the numbers are not

Reply:
The original names are only informative for the back-end they were taken from. For any other back-end these names don't make sense.
Also the index 2 was missing.
In fact, address spaces are just numbers, so I propose removing this enum altogether and only using u32 instead.

Reply:
Oh okay, been reading up on this and that seems to check out